stumpy/core.py

```diff
@@ -2254,15 +2349,108 @@ def rolling_isconstant(a, w):
-        Rolling window isconstant.
+    T_subseq_isconstant : numpy.ndarray
+        Rolling window isconstant
+    """
+    if custom is None:
+        custom = _rolling_isconstant
+
+    if callable(custom):
+        custom_args = []
+        for arg_name, arg in inspect.signature(custom).parameters.items():
+            if arg.default == inspect.Parameter.empty:
+                custom_args.append(arg_name)
+
+        incomp_args = set(custom_args).difference({"a", "w"})
+        if len(incomp_args) > 0:  # pragma: no cover
+            msg = (
+                f"Incompatible arguments {incomp_args} found in `custom_func`. "
+                + "Please provide a `custom_func` with arguments `a`, a 1-D array, "
+                + "and `w`, the window size."
+            )
+            raise ValueError(msg)
+
+        axis = a.ndim - 1
+        T_subseq_isconstant = np.apply_along_axis(
+            lambda a_row, w: custom(a_row, w), axis=axis, arr=a, w=w
+        )
+
+    elif isinstance(custom, np.ndarray):
+        T_subseq_isconstant = custom
+
+    else:  # pragma: no cover
+        msg = (
+            "The `custom` must be of type `np.ndarray` or a callable object. "
+            + f"Found {type(custom)} instead."
+        )
+        raise ValueError(msg)
+
+    if not issubclass(T_subseq_isconstant.dtype.type, np.bool_):  # pragma: no cover
+        msg = (
+            f"The dtype of `T_subseq_isconstant` is {T_subseq_isconstant.dtype} "
+            + "but dtype `np.bool` was expected"
+        )
+        raise ValueError(msg)
+
+    return T_subseq_isconstant
+
+
+def fix_isconstant_isfinite_conflicts(
+    T, m, T_subseq_isconstant, T_subseq_isfinite=None
+):
     """
-    axis = a.ndim - 1
-    return np.apply_along_axis(
-        lambda a_row, w: _rolling_isconstant(a_row, w), axis=axis, arr=a, w=w
-    )
+    Fix `T_subseq_isconstant` by setting its element to False if their
+    corresponding value in `T_subseq_isfinite` is False.
+
+    Parameters
+    ----------
+    T : numpy.ndarray
+        Time series
+
+    m : int
+        Subsequence window size
+
+    T_subseq_isconstant : numpy.ndarray
+        A numpy array `dtype` of boolean that indicates whether a subsequence
+        is constant (True)  or not (False).
+
+        T_subseq_isfinite : numpy.ndarray, default None
+        A boolean array that indicates whether a subsequence in `T` contains a
+        `np.nan`/`np.inf` value (False)
+
+    Returns
+    -------
+    fixed : numpy.ndarray
+        The same as input `T_subseq_isconstant` but with indices set to False
+        if their corresponding subsequence are not finite.
+    """
+    if T_subseq_isfinite is None:
+        T_subseq_isfinite = rolling_isfinite(T, m)
+
+    fixed = np.logical_and(T_subseq_isconstant, T_subseq_isfinite)
+
+    conflicts = fixed != T_subseq_isconstant
+    if np.any(conflicts):
+        msg = (
+            "Found indices where T_subseq_isconstant is True for subsequence "
```

Comment:
I think this message can be more concise with something like:

f"Subsequences located at indices {np.nonzero(conflicts)} contain one or more `np.nan`/`np.inf` and so their corresponding values in `T_subseq_isconstant` have been automatically switched from `True` to `False`.

Reply:
Thanks for the feedback. I will revise it,

Reply:
It is already added to the unit test. In fact, that resulted in raising an error, which in turn helped me with solving it.